lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -792,8 +782,234 @@ namespace Js
+            auto getGroup = [&](int captureIndex, Var nonMatchValue) {
+                return captureIndex <= numberOfCaptures ? captures[captureIndex] : nonMatchValue;
+            };
+            UnifiedRegex::GroupInfo match(position, matchStr->GetLength());
+            int numGroups = numberOfCaptures + 1; // Take group 0 into account.
+            ReplaceFormatString(
+                scriptContext,
+                numGroups,
+                getGroup,
+                input,
+                matchStr->GetString(),
+                match,
+                replace,
+                substitutions,
+                substitutionOffsets,
+                resultBuilder);
+        };
+        return RegexEs6ReplaceImpl(scriptContext, thisObj, input, appendReplacement, noResult);
+    }
+
+    Var RegexHelper::RegexEs6ReplaceImpl(ScriptContext* scriptContext, RecyclableObject* thisObj, JavascriptString* input, JavascriptFunction* replaceFn)
+    {
+        auto appendReplacement = [&](
+            CompoundString::Builder<64 * sizeof(void *) / sizeof(wchar_t)>& resultBuilder,
+            ArenaAllocator* tempAlloc,
+            JavascriptString* matchStr,
+            int numberOfCaptures,
+            Var* captures,
+            CharCount position)
+        {
+            // replaceFn Arguments:
+            //
+            // 0: this
+            // 1: matched
+            // 2: capture1
+            // ...
+            // N + 1: capture N
+            // N + 2: position
+            // N + 3: input
+
+            // Number of captures can be at most 99, so we won't overflow.
+            ushort argCount = (ushort) numberOfCaptures + 4;
+
+            PROBE_STACK(scriptContext, argCount * sizeof(Var));
+            Var* args = (Var*) _alloca(argCount * sizeof(Var));
+
+            args[0] = scriptContext->GetLibrary()->GetUndefined();
+#pragma prefast(suppress:6386, "The write is within the bounds")
+            args[1] = matchStr;
+            for (int i = 1; i <= numberOfCaptures; ++i)
+            {
+                args[i + 1] = captures[i];
+            }
+            args[numberOfCaptures + 2] = JavascriptNumber::ToVar(position, scriptContext);
+            args[numberOfCaptures + 3] = input;
+
+            JavascriptString* replace = JavascriptConversion::ToString(
+                replaceFn->CallFunction(Arguments(CallInfo(argCount), args)),
+                scriptContext);
+
+            resultBuilder.Append(replace);
+        };
+        return RegexEs6ReplaceImpl(scriptContext, thisObj, input, appendReplacement, /* noResult */ false);
+    }
+
+    template<typename ReplacementFn>
+    Var RegexHelper::RegexEs6ReplaceImpl(ScriptContext* scriptContext, RecyclableObject* thisObj, JavascriptString* input, ReplacementFn appendReplacement, bool noResult)
+    {
+        bool global = JavascriptRegExp::GetGlobalProperty(thisObj, scriptContext);
+        bool unicode = false; // Dummy value. It isn't used below unless "global" is "true".
```

Comment:
To me, _default_ would mean it'll be used. However, `unicode` isn't used at all anywhere in this function unless `global` is `true`. I just picked `false` to initialize `unicode` and wanted to mention why :wink: